L1Trigger/TrackFindingTracklet/src/VMRouterCM.cc

```diff
@@ -344,17 +344,43 @@ void VMRouterCM::execute(unsigned int) {
         }
 
         int lutval = -999;
+        int stub_rbin = 0;
 
         if (inner > 0) {
           if (layerdisk_ < N_LAYER) {
             lutval = (!isTripletSeed ? melut : melutOld);
           } else {
             if (inner == 2 && iseed == Seed::L2L3D1) {
               lutval = 0;
-              if (stub->rvalue() < 10) {
-                lutval = 8 * (1 + (stub->rvalue() >> 2));
+              if (stub->rvalue() < 10 || (stub->rvalue() > settings_.rmindiskl2overlapvm() /
```

Comment:
A general comment -- If you've understood any obscure code, please add a comment to explain it. e.g. Here, I'd naively guess rvalue is the radius of the stub, but requiring this to be less than 10cm makes no sense. And is 10 a magic hard-wired number?

Reply:
I've added an explanation in the code (rvalue for the 2S stubs is not the radius in cm but an index to the ring), and changed "10" to "2*N_DSS_MOD", which is the number of rings in the endcaps.